@xen-orchestra/proxy/src/app/mixins/backups/index.js

```diff
@@ -133,6 +133,28 @@ export default class Backups {
 
     app.api.addMethods({
       backup: {
+        deleteVmBackup: [
+          ({ filename, remote }) =>
+            using(app.remotes.getAdapter(remote), async adapter => {
+              const metadata = JSON.parse(String(await adapter.handler.readFile(filename)))
+              metadata._filename = filename
+
+              if (metadata.mode === 'delta') {
+                await adapter.deleteDeltaVmBackups([metadata])
+              } else if (metadata.mode === 'full') {
+                await adapter.deleteFullVmBackups([metadata])
+              } else {
+                throw new Error(`no deleter for backup mode ${metadata.mode}`)
+              }
```

Comment:
Should be abstracted in the adapter.

Reply:
Why?

Reply:
Because this logic is completely unnecessary here.

We're going to move all this code in the `@xen-orchestra/backups` lib and use it both in the proxy and `xo-server`, it does not make sense to duplicate this logic.